pkg/netpol/connlist/exposure_map.go

```diff
@@ -52,11 +47,17 @@ func (ex exposureMap) addPeerUnprotectedData(pe *eval.PolicyEngine, peer Peer, i
 	if err != nil {
 		return false, err
 	}
-	if !protected {
-		if _, ok := ex[peer]; !ok {
-			ex.initiatePeerEntry(peer)
+	_, ok := ex[peer]
+	if !ok && !protected {
+		err = ex.initiateEmptyPeerEntry(pe, peer)
+		return true, err
+	}
+	if ok && protected {
+		if isIngress {
```

Comment:
func `addPeerUnprotectedData` checks if the peer is protected on the given direction:
1- (!ok && !protected) : if it is not protected and it is not in the map already : we add a new empty entry (which means with false protected flags for both directions)- if we have at least one direction which the peer is not protected on; this peer should be in the map (an exposure-analysis case).
2- (ok && protected) if it is protected and there is an entry in the map : so we will update the flag of the current direction as protected 
(might be a case that the peer is not protected on ingress - so an empty entry was created first but called now with egress and got protected - so will update the "egress" unprotected flag to protected )

3. (protected && !ok ) : if it is not in the map and we get protected, we will skip at this stage (cause it might be protected and exposed to only known end-points - which means not exposure-analysis case )

4. (ok && !protected) if ok and not protected: its already in the map with the right data (no need to update ): 
the `ok && protected`  can be replaced with just `ok` (as 4 doesn't change its data)

Reply:
> we add a new empty entry (which means with false protected flags for both directions)- if we have at least one direction which the peer is not protected on

so this means that one entry could be initialized incorrectly as `false` ?
maybe the value in this entry should have 3 options: (false/true/unknown)?

Reply:
exactly, will change to be with 3 values

Reply:
committed changes